--- a/mfhdf/Makefile.in
+++ b/mfhdf/Makefile.in
@@ -377,8 +377,8 @@
 @HDF_BUILD_FORTRAN_TRUE@FORTRAN_DIR = fortran
 
 # src folder in root Makefile, build other folders now
-SUBDIRS = $(XDR_DIR) libsrc $(FORTRAN_DIR) test ncdump ncgen dumper hdfimport hdiff hrepack nctest
-DIST_SUBDIRS = xdr libsrc fortran test ncdump ncgen dumper hdfimport hdiff hrepack nctest examples
+SUBDIRS = $(XDR_DIR) libsrc $(FORTRAN_DIR) test ncdump dumper hdfimport hdiff hrepack nctest
+DIST_SUBDIRS = xdr libsrc fortran test ncdump dumper hdfimport hdiff hrepack nctest examples
 all: all-recursive
 
 .SUFFIXES:
